### PR TITLE
Feature/display discount order summary

### PR DESCRIPTION
**Pull request by huangsri**
# Overview
- Display discount on Summary component

# Preview
Screen Shot 2564-01-12 at 08 33 15

### CODE REVIEW

File: packages/payment/src/components/Summary/Summary.tsx [hunk 1]
```diff
@@ -94,6 +103,42 @@ export const Summary = (props: SummaryProps) => {
...
+          <Box pt="2" pb="2">
+            {discount?.vat}
+          </Box>
+          <Box pt="2" pb="2" color="red.400" fontWeight="600">
+            {discount?.value}
+          </Box>
+        </React.Fragment>
+      )
+    }
+  }
```
Comment (mypum) [inline packages/payment/src/components/Summary/Summary.tsx:140]: You might separate as two component
1. Show discount for single value
2. Show discounts in many

Render condition
`
        {
          isArray(discount) ? <Disccounts discounts={discount} /> : <Discount discount={discount}>
        }
`

Discounts component will be like this
```
const Discounts = ({ discounts }: { discounts: Discount[] }) => {
  return (
    <Fragment>
      {discounts.map((discount, idx) => (
        <Fragment key={idx}>
          <Box textAlign="left" pt="2" pb="2">
            {discount.label}
          </Box>
          <Box pt="2" pb="2">
            {discount?.vat}
          </Box>
          <Box pt="2" pb="2" color="red.400" fontWeight="600">
            {discount.value}
          </Box>
        </Fragment>
      ))}
    </Fragment>
  )
}
```

or you might separate renderDiscount into one component and implement condition render in there instead.

File: packages/payment/src/components/Summary/Summary.stories.tsx [hunk 2]
```diff
@@ -58,6 +58,20 @@ export const individual = () => {
...
+export const withDiscount = () => {
+  return (
+    <Summary
+      name="Small Package"
+      description="ระยะเวลา 12 เดือน"
+      net="3,200.00"
+      vatPercent="7%"
+      vat="224.00"
+      total="3,424.00"
+      discount={{ label: 'ส่วนลดมูลค่าคงเหลือแพคเกจ', value: '300.00' }}
```
Comment (mypum) [inline packages/payment/src/components/Summary/Summary.stories.tsx:70]: Be better if you add more example for array of discount

File: packages/payment/src/components/Summary/Summary.tsx (hunk 1 above)
Comment (huangsri) [reply packages/payment/src/components/Summary/Summary.tsx:140]: Great Idea!!

**Comment by huangsri** [comment]
@mypum Please have a look again <3

**Review by mypum — APPROVED** [review]

**Pull request merged**